pkg/operator2/oauth.go

```diff
@@ -136,3 +148,172 @@ func (c *osinOperator) handleOAuthConfig(configOverrides []byte) (*corev1.Config
+			URL:    ldapConfig.URL,
+			BindDN: ldapConfig.BindDN,
+			BindPassword: configv1.StringSource{
+				StringSourceSpec: configv1.StringSourceSpec{
+					// TODO: this could be Env, Value, File, either can be encrypted
+					File: getLocalSecretFilename(ldapConfig.BindPassword),
+				},
+			},
+			Insecure: ldapConfig.Insecure,
+			CA:       getLocalConfigMapFilename(ldapConfig.CA),
+		}
+		serializer.Encode(&p, bytesWriter)
+
+	case configv1.IdentityProviderTypeKeystone:
+		keystoneConfig := providerConfig.Keystone
+		p := osinv1.KeystonePasswordIdentityProvider{
+			RemoteConnectionInfo: configv1.RemoteConnectionInfo{
+				URL: keystoneConfig.URL,
+				CA:  getLocalConfigMapFilename(keystoneConfig.CA),
+				CertInfo: configv1.CertInfo{
+					CertFile: getLocalSecretFilename(keystoneConfig.TLSClientCert),
+					KeyFile:  getLocalSecretFilename(keystoneConfig.TLSClientKey),
+				},
+			},
+			DomainName:          keystoneConfig.DomainName,
+			UseKeystoneIdentity: !keystoneConfig.LegacyLookupByUsername,
+		}
+		serializer.Encode(&p, bytesWriter)
+
+	case configv1.IdentityProviderTypeRequestHeader:
+		requestHeaderConfig := providerConfig.RequestHeader
+		p := osinv1.RequestHeaderIdentityProvider{
+			LoginURL:                 requestHeaderConfig.LoginURL,
+			ChallengeURL:             requestHeaderConfig.ChallengeURL,
+			ClientCA:                 getLocalConfigMapFilename(requestHeaderConfig.ClientCA),
+			ClientCommonNames:        requestHeaderConfig.ClientCommonNames,
+			Headers:                  requestHeaderConfig.Headers,
+			PreferredUsernameHeaders: requestHeaderConfig.PreferredUsernameHeaders,
+			NameHeaders:              requestHeaderConfig.NameHeaders,
+			EmailHeaders:             requestHeaderConfig.EmailHeaders,
+		}
+		serializer.Encode(&p, bytesWriter)
+
+	case configv1.IdentityProviderTypeGitHub:
+		githubConfig := providerConfig.GitHub
+		p := osinv1.GitHubIdentityProvider{
+			ClientID: githubConfig.ClientID,
+			ClientSecret: configv1.StringSource{
+				// TODO: this could be Env, Value, File, either can be encrypted
+				StringSourceSpec: configv1.StringSourceSpec{
+					File: getLocalSecretFilename(githubConfig.ClientSecret),
+				},
+			},
+			Organizations: githubConfig.Organizations,
+			Hostname:      githubConfig.Hostname,
+			CA:            getLocalConfigMapFilename(githubConfig.CA),
+		}
+		serializer.Encode(&p, bytesWriter)
+
+	case configv1.IdentityProviderTypeGitLab:
+		gitlabConfig := providerConfig.GitLab
+		p := osinv1.GitLabIdentityProvider{
+			CA:       getLocalConfigMapFilename(gitlabConfig.CA),
+			URL:      gitlabConfig.URL,
+			ClientID: gitlabConfig.ClientID,
+			ClientSecret: configv1.StringSource{
+				// TODO: this could be Env, Value, File, either can be encrypted
+				StringSourceSpec: configv1.StringSourceSpec{
+					File: getLocalSecretFilename(gitlabConfig.ClientSecret),
+				},
+			},
+			Legacy: &gitlabConfig.LegacyOAuth2, // TODO: better mapping of 2 to 3 values?
```

Comment:
I guess this was turned into a pointer in OSIN config for a reason but there is probably no way to reach the `nil` value?

Reply:
Hard code this to `new(bool)` and add a comment saying `we require OIDC for GitLab now`